app/models/package_manager/pypi/version_processor.rb

```diff
@@ -11,20 +11,31 @@ def initialize(project_releases:, project_name:, known_versions:)
 
       def execute
         @project_releases.map do |project_release|
-          next @known_versions[project_release.version_number] if @known_versions.key?(project_release.version_number)
-
-          version_number = project_release.version_number
-          original_license = json_api_single_release_for_version(version_number).license
-          rss_api_release = rss_api_releases_hash[version_number]
-          published_at = project_release.published_at || rss_api_release&.published_at
-          yanked = project_release&.yanked? || false
-
-          {
-            number: version_number,
-            published_at: published_at,
-            original_license: original_license,
-            yanked: yanked,
-          }
+          if @known_versions.key?(project_release.version_number)
```

Comment:
This is basically turning the postfix if into a full if/else statement, with the case if we already know the version only potentially changing the status field.